src/ts/components/shared/chat-log/chat-log.ts

```
@@ -414,6 +450,164 @@ export class ChatLog implements AfterViewInit, OnDestroy, DoCheck {
 
 		return index;
 	}
+	filterChat() {
+		this.clearTimeOutAutoClear();
+		this.clearTimeOutAutoUnfocus();
+		let value = this.filterTab.nativeElement.value;
+		if (!value) {
+			this.clearTimeOutAutoClear();
+			this.unFocus();
+			this.showHideChatLogLines('', false);
+			return;
+		}
+		else this.filterColor = this.bg;
+
+		this.autoClear = setTimeout(() => {
+			this.filterTab.nativeElement.value = '';
+			this.filterChat();
+			this.unFocus();
+		}, SECOND * 30);
+
+		this.autoUnfocus = setTimeout(() => {
+			this.unFocus();
+		}, SECOND * 3);
+
+		let toLowerCase = false;
+
+		if (value.startsWith('#')) value = value.slice(1);
+		else if (value.startsWith('/')) {
+			value = value.slice(1);
+			try {
+				const regExp = new RegExp(value);
+				this.showHideChatLogLines(regExp, toLowerCase);
+				this.filterTab.nativeElement.style.color = '';
+			} catch (err) {
```

Why are we just swallowing exceptions here? Either make a comment, or do something about it